### PR TITLE
fix-markdown-numbered-lists

### PR DESCRIPTION
**Pull request by JeanPaulSG**
## Test plan

@blvckcoffee or @dulanism to check formatting of code blocks and numbered lists please.

### CODE REVIEW

**Comment by JeanPaulSG** [comment]
FYI @caugustus-sourcegraph I corrected some markdown formatting errors.

File: doc/admin/deploy/docker-compose/index.md [hunk 1]
```diff
@@ -47,55 +51,56 @@ The [`sourcegraph/deploy-sourcegraph-docker`](https://github.com/sourcegraph/dep
...
 
-4. Remove your local bare clone. 
-  ```bash
+4\. Remove your local bare clone. 
+
+```bash
   cd ..
-  rm -rf deploy-sourcegraph-docker.git
-  ```
+  m -rf deploy-sourcegraph-docker.git
```
Comment (ryanslade) [inline doc/admin/deploy/docker-compose/index.md:77]: Should be `rm` I think

**Review by ryanslade — APPROVED** [review]
Looks good, I think. 

I've not seen the `\` in numbered lists before though, why is that necessary?

**Comment by JeanPaulSG** [comment]
> Looks good, I think.
> 
> I've not seen the `\` in numbered lists before though, why is that necessary?

Numbered list was restarting from 1 after every code block, so I had to escape the list format and number the lines explicitly.

**Comment by sanderginn** [comment]
@JeanPaulSG FYI I had a quick peek at this PR because I thought this might have fixed something I had pushed recently (not the case). The numbering is reset due to the (lack of) indentation of the code blocks. If you prefix a codeblock in an itemization/enumeration with three spaces on each line, the block is indented and grouped with the current item, and the next item will continue counting in case of an enumeration. Typical Markdown headaches :-)

**Example without three spaces:**
1. Stop the deployment, and restart the databases service only to ensure there are no other connections during backup and restore.

```bash
docker-compose down
docker-compose -f db-only-migrate.docker-compose.yaml up -d
```

1. Generate the database dumps

```bash
docker exec pgsql sh -c 'pg_dump -C --username sg sg' > sourcegraph_db.out
docker exec codeintel-db -c 'pg_dump -C --username sg sg' > codeintel_db.out
```

1. Ensure the `sourcegraph_db.out` and `codeintel_db.out` files are moved to a safe and secure location. 


**Example with three spaces:**
1. Stop the deployment, and restart the databases service only to ensure there are no other connections during backup and restore.

   ```bash
   docker-compose down
   docker-compose -f db-only-migrate.docker-compose.yaml up -d
   ```

1. Generate the database dumps

   ```bash
   docker exec pgsql sh -c 'pg_dump -C --username sg sg' > sourcegraph_db.out
   docker exec codeintel-db -c 'pg_dump -C --username sg sg' > codeintel_db.out
   ```

1. Ensure the `sourcegraph_db.out` and `codeintel_db.out` files are moved to a safe and secure location.

**Comment by JeanPaulSG** [comment]
Thanks @sanderginn for looking into this, I ran into that solution when I was researching this issue. I just double checked it now: It works for the previewer but when rendered in live or built locally it restarts the numbering... See pic: Left is local build, right is preview, centre panel is code with three spaces in code blocks as per your suggestion. I wonder if the markdown implementation we use is somehow out of date? 
image

**Pull request merged**